### PR TITLE
Card component; AutoImage enhancement; Podcast Screen updates.

### PR DESCRIPTION
**Pull request by yulolimum**
## Please verify the following:

- [x] `yarn ci:test` **jest** tests pass with new tests, if relevant
- [x] `README.md` has been updated with your changes, if relevant

#### Card Component:
- Adds a new Card component.
- Adds DemoCard to showcase.
- Updates DemoPodcastListScreen with new Card.

#### AutoImage Enhancements:
- Adds new feature to AutoImage to auto-scale images to a provided max height/width while retaining aspect ratio.
- Update DemoAutoImage with examples.

## Proofs

| Card Component Showcase | AutoImage Enhancement Demo | Podcast Screen Updates |
|---|---|---|
| yulolimum-capture-2022-09-02--17-56-02 | yulolimum-capture-2022-09-02--17-57-39 | yulolimum-capture-2022-09-02--17-58-12 |
|  |  |  |
|  |  |  |
|  |  |  |
|  |  |  |
|  |  |  |

### CODE REVIEW

File: boilerplate/app/components/AutoImage.tsx [hunk 1]
```diff
@@ -1,26 +1,74 @@
...
+    const aspectRatio = Math.min(maxWidth / remoteWidth, maxHeight / remoteHeight)
+    return [remoteWidth * aspectRatio, remoteHeight * aspectRatio]
+  } else if (maxWidth) {
+    return [maxWidth, maxWidth / remoteAspectRatio]
+  } else if (maxHeight) {
+    return [maxHeight * remoteAspectRatio, maxHeight]
+  } else {
+    return [remoteWidth, remoteHeight]
+  }
+}
```
Comment (yulolimum) [inline boilerplate/app/components/AutoImage.tsx:54]: This hook adds ability for automatic image scaling. This is integrated into AutoImage for remote image scaling, but can be updated to include the same functionality for local image as well (needs to call different method to get the size).

File: boilerplate/app/models/Episode.ts [hunk 2]
```diff
@@ -30,6 +30,17 @@ export const EpisodeModel = types
...
+      const defaultValue = { title: episode.title?.trim(), subtitle: "" }
+
+      if (!defaultValue.title) return defaultValue
+
+      const titleMatches = defaultValue.title.match(/^(RNR.*\d)(?: - )(.*$)/)
+
+      if (!titleMatches || titleMatches.length !== 3) return defaultValue
+
+      return { title: titleMatches[1], subtitle: titleMatches[2] }
+    },
```
Comment (yulolimum) [inline boilerplate/app/models/Episode.ts:43]: Adds feature to split title into two chunks (ep number and description). Has fallbacks if the title doesn't contain the correct pattern.

**Review by jamonholmgren — APPROVED** [review]
Looks epic. BTW, I was having trouble with the AutoImage, thanks for fixing that.

**Review by mazenchami — APPROVED** [review]
LGTM! 🎉

**Review by jamonholmgren — APPROVED** [review]

**Pull request merged**